document-store/src/main/java/org/hypertrace/core/documentstore/mongo/update/parser/MongoIncrementOperationParser.java

```diff
@@ -0,0 +1,24 @@
+package org.hypertrace.core.documentstore.mongo.update.parser;
+
+import static org.hypertrace.core.documentstore.model.subdoc.UpdateOperator.INCREMENT;
+import static org.hypertrace.core.documentstore.mongo.MongoUtils.merge;
+
+import com.mongodb.BasicDBObject;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import org.hypertrace.core.documentstore.model.subdoc.UpdateOperator;
+
+@AllArgsConstructor
+public class MongoIncrementOperationParser extends MongoOperationParser {
+  private static final String INCREMENT_CLAUSE = "$inc";
+
+  @Override
+  UpdateOperator operator() {
+    return INCREMENT;
+  }
+
+  @Override
+  BasicDBObject wrapWithOperator(List<BasicDBObject> parsed) {
```

Comment:
TBH, I am not clear what is happening here. This was a abstract method and all other implementation were using merge operation on parsed.

Reply:
I think that's fine. 
It is to handle multi-value cases.
Basically, if you send a SubDocumentValue.ofNumbers(...), it would add all those numbers to that field.
Let's leave it as is for now.